torch/_refs/__init__.py

```diff
@@ -3649,6 +3649,17 @@ def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorL
         else:
             return _a
 
+    if a.is_contiguous() and a.device.type != "xla":
+        torch._check(a.numel() > 0)
```

Comment:
Current eager allows to work with 0 numel tensors in the way, we should allow a.numel() == 0
e.g.
```
import torch
t = torch.tensor([])
t.shape
>> torch.Size([0])
t.numel()
>> 0
r = t.view(0, 10)
r.shape
>> torch.Size([0, 10])
```

Reply:
The case ```a.numel() == 0``` is already covered earlier in the code on line 3627.